test/functional/test_framework/test_framework.py

```diff
@@ -103,6 +103,11 @@ def __init__(self) -> None:
         self.supports_cli = True
         self.bind_to_localhost_only = True
         self.parse_args()
+        # Note that we haven't called setmocktime on any nodes yet. If a test wants to use
+        # setmocktime, it should call bumpmocktime(0) at the beginning of run_test in order to
+        # prevent spurious failures. If the test does not want to use setmocktime, it must not call
+        # setmocktime because the time will not move forward afterwards.
+        self.mocktime = int(time.time())
```

Comment:
Wouldn't it make more sense to have `bumpmocktime` on TestNode and set `self.mocktime` when ever `setmocktime` is called?

* allows for node independent mocktime bumping
* allows to use an initial mocktime other than time.time()

Reply:
Good point, moved from TestFramework to TestNode

Reply:
Added a wrapper for the RPC to update `TestNode::mocktime` every time `setmocktime` is called.